dartagnan/src/main/java/com/dat3m/dartagnan/encoding/PropertyEncoder.java

```diff
@@ -204,8 +205,32 @@ private BooleanFormula encodeLastCoConstraints() {
                     BooleanFormula sameAddress = context.sameAddress(init, w1);
                     Formula v2 = ExpressionEncoder.getLastMemValueExpr(init.getBase(), init.getOffset(), fmgr);
                     BooleanFormula sameValue = context.equal(context.value(w1), v2);
-                    enc.add(bmgr.implication(bmgr.and(lastCoExpr, sameAddress), sameValue));
+                    if (Arch.coIsTotal(program.getArch())) {
+                        enc.add(bmgr.implication(bmgr.and(lastCoExpr, sameAddress), sameValue));
+                    }
```

Comment:
You can move the check upwards into the first conditional: `if (doEncodeFinalAddrssValues && Arch.coIsTotal(program.getArch()))`.
This is because all the code inbetween the two conditionals does nothing if `Arch.coIsTotal(program.getArch())` fails to hold.

Reply:
Updated